app/models/user.rb

```diff
@@ -7,4 +8,29 @@ class User < ApplicationRecord
                   uniqueness: {case_sensitive: false}
   validates :password, presence: true, length: {minimum: Settings.digit_6}
   has_secure_password
+
+  def self.digest string
+    min_cost = BCrypt::Engine::MIN_COST
+    cost = ActiveModel::SecurePassword.min_cost ? min_cost : BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
```

Comment:
```suggestion
self << class
  def digest string
    min_cost = BCrypt::Engine::MIN_COST
    cost = ActiveModel::SecurePassword.min_cost ? min_cost : BCrypt::Engine.cost
    BCrypt::Password.create(string, cost: cost)
  end

  def new_token
    SecureRandom.urlsafe_base64
  end
  end
```
nên dùng cú pháp này để nhóm các class method lại nhé

Reply:
em đã gộp class_method vào class << self  rồi ạ

Reply:
fixx